File: selfstudy/python/words_min.py
'''
task 193:
Words - Min
You need to find the minimum value among word lengths
'''

def lowest_words(lengths):
    return min(lengths) if lengths else 0

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], lowest_words([10, 20, 30])),
        ([50, 60, 70], lowest_words([50, 60, 70])),
        ([1, 2, 3], lowest_words([1, 2, 3])),
        ([100, 200], lowest_words([100, 200])),
        ([], lowest_words([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = lowest_words(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# done

File: selfstudy/golang/words_min.go
/*
task 293:
Words - Min
You need to find the minimum value among word lengths
*/

package main

import (
	"fmt"
	"reflect"
)

func LowestWords(lengths []int) int {
	if len(lengths) == 0 { return 0 }
	m := lengths[0]
	for _, v := range lengths {
		if v < m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, LowestWords([]int{10, 20, 30})},
		{[]int{50, 60, 70}, LowestWords([]int{50, 60, 70})},
		{[]int{1, 2, 3}, LowestWords([]int{1, 2, 3})},
		{[]int{100, 200}, LowestWords([]int{100, 200})},
		{[]int{}, LowestWords([]int{})},
	}
	
	for idx, t := range tests {
		res := LowestWords(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
